plugins/modules/apt_rpm.py

```diff
@@ -118,6 +124,23 @@
 UPDATE_KERNEL_ZERO = "\nTry to install new kernel "
 
 
+def local_rpm_package_name(path):
+    """return package name of a local rpm passed in.
+    Inspired by ansible.builtin.yum"""
+
+    ts = rpm.TransactionSet()
+    ts.setVSFlags(rpm._RPMVSF_NOSIGNATURES)
+    fd = os.open(path, os.O_RDONLY)
+    try:
+        header = ts.hdrFromFdno(fd)
+    except rpm.error as e:
+        return None
+    finally:
+        os.close(fd)
+
+    return header[rpm.RPMTAG_NAME].decode()
```

Comment:
Calling `decode()` likely won't work with Python 2.

Reply:
Thanks for pointing this out

I'm not familiar enough with Python 2, but for now it looks like 2.7 supports the `decode()` method:
https://docs.python.org/2/library/stdtypes.html#str.decode

Also on my 2.7.18,
- `header[rpm.RPMTAG_NAME]` evaluates to `'whatever'`
- `header[rpm.RPMTAG_NAME].decode()` evaluates to `u'class-screencast-serve'`
- finally, `"%s -q --provides %s" % (RPM_PATH, name)` evaluates to `u'/usr/bin/rpm -q --provides whatever'`

Does the `module.run_command()` argument have to be str instead of unicode?

Reply:
I guess `str` is probably better. If you replace `decode()` with a call to `to_native` (`from ansible.module_utils.common.text.converters import to_native`), then you'll end up with `str` on both Python 2 and 3.